packages/client/components/ActivityLibrary/ActivityDetailsSidebar.tsx

```diff
@@ -136,12 +147,27 @@ const ActivityDetailsSidebar = (props: Props) => {
   } = useDialogState()
 
   const [selectedUsers, setSelectedUsers] = React.useState<Option[]>([])
+  const selectedUser = selectedUsers[0]
+  const {organizations: viewerOrganizations} = viewer
+
+  const viewerOrganizationIds = new Set(viewerOrganizations.map((org) => org.id))
+
+  const selectedUserOrganizationIds = selectedUser?.organizationIds ?? []
+  const mutualOrgs = selectedUserOrganizationIds.filter((orgId) => viewerOrganizationIds.has(orgId))
+
+  const showOrgPicker = selectedUser && (mutualOrgs.length > 1 || !mutualOrgs.length)
+
+  const manuallySelectedOrgId = showOrgPicker && selectedOrgId !== '' && selectedOrgId
+  const firstMutualOrgId = mutualOrgs[0]
+  const defaultOrgId = selectedTeam.orgId
+  const newOneOnOneTeamOrgId = manuallySelectedOrgId
+    ? manuallySelectedOrgId
+    : firstMutualOrgId ?? defaultOrgId
```

Comment:
-1 This logic is a bit dense, and I think there's some opportunity to simplify here. Here's a general approach that I'd recommend:
* Determine the default org ID (that is, `mutualOrgsIds[0] ?? selectedTeam.orgId`) and use that as initial state for `selectedOrgId`
* When mutual orgs change, determine a new default `selectedOrgId` iff the current `selectedOrgId` is _not_ in the new mutual orgs array

This way, `selectedOrgId` is clearly the single source of truth for which org is currently selected, and we still update it automatically if it needs to be changed.